### PR TITLE
[ticket/12630] Add debug.url_matcher

### PR DESCRIPTION
**Pull request by rubencm**
PHPBB3-12630

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-12630

### CODE REVIEW

File: phpBB/phpbb/routing/router.php [hunk 1]
```diff
@@ -200,7 +200,7 @@ protected function create_dumped_url_matcher()
 	{
 		try
 		{
-			$cache = new ConfigCache("{$this->cache_dir}url_matcher.{$this->php_ext}", defined('DEBUG'));
+			$cache = new ConfigCache("{$this->cache_dir}url_matcher.{$this->php_ext}", $this->container->getParameter('debug.url_matcher'));
```
Comment (Nicofuma) [inline phpBB/phpbb/routing/router.php]: please inject the parameter instead of using $this->container

**Comment by rubencm** [comment]
@Nicofuma I have added the parameter with value false, don't know if is correct or better it should be true

**Comment by Nicofuma** [comment]
in tests? better use true imo

**Comment by Nicofuma** [comment]
but both should be good

**Comment by rubencm** [comment]
@marc1706 Please, check this too, is very similar to #5335

**Review by marc1706 — APPROVED** [review]

**Pull request merged**